src/Google.Api.Gax/Api/Gax/RetrySettings.cs

```diff
@@ -0,0 +1,155 @@
+        public IJitter DelayJitter { get; set; } = RandomJitter;
+
+        /// <summary>
+        /// Creates a deep clone of this instance. No validation is performed.
+        /// It is assumed that the delay jitter does not require deep cloning.
+        /// </summary>
+        /// <returns>A deep clone of this instance.</returns>
+        public RetrySettings Clone() =>
+            new RetrySettings
+            {
+                RetryBackoff = RetryBackoff?.Clone(),
+                TimeoutBackoff = TimeoutBackoff?.Clone(),
+                DelayJitter = DelayJitter,
+                RetryFilter = RetryFilter
+            };
+
+        /// <summary>
+        /// Validates the settings.
+        /// </summary>
+        /// <returns>A reference to this instance, for use in chaining calls.</returns>
+        /// <param name="paramName">The parameter name to use in any exception thrown.</param>
+        /// <exception cref="ArgumentException">The settings are invalid.</exception>
+        public RetrySettings Validate(string paramName)
+        {
+            GaxPreconditions.CheckArgument(RetryBackoff != null, paramName, "No retry backoff specified");
+            GaxPreconditions.CheckArgument(TimeoutBackoff != null, paramName, "No timeout backoff specified");
+            GaxPreconditions.CheckArgument(DelayJitter != null, paramName, "No delay jitter specified");
+            GaxPreconditions.CheckArgument(RetryFilter != null, paramName, "No retry filter specified");
+            return this;
+        }
+
+        /// <summary>
+        /// Provides a mechanism for applying jitter to delays between retries.
+        /// See the <see cref="DelayJitter"/> property for more information.
+        /// </summary>
+        public interface IJitter
+        {
+            /// <summary>
+            /// Returns the actual delay to use given a maximum available delay.
+            /// </summary>
+            /// <param name="maxDelay">The maximum delay provided by the backoff settings</param>
+            /// <returns>The delay to use before retrying.</returns>
+            TimeSpan GetDelay(TimeSpan maxDelay);
+        }
+
+        /// <summary>
+        /// The default jitter which returns a uniformly distributed random delay between 0 and
+        /// the specified maximum.
+        /// </summary>
+        public static IJitter RandomJitter { get; } = new RandomJitterImpl();
+
+        /// <summary>
+        /// A jitter which simply returns the specified maximum delay.
+        /// </summary>
+        public static IJitter NoJitter { get; } = new NoJitterImpl();
+
+        // TODO: Is this a reasonable default? Does it make sense to have a default at all?
+
+        /// <summary>
+        /// The default retry filter, which retries operations which fail due to a status code of "not found".
+        /// </summary>
+        public static Predicate<RpcException> DefaultFilter { get; } = FilterForStatusCodes(StatusCode.NotFound);
```

Comment:
I thought there was also a status code that meant try again or something like that. Not found I guess it means 404? That doesn't seem like a good general filter though, if a file doesn't exist it doesn't necessary mean that it is going to exist later.